bot.py

```diff
@@ -322,9 +458,26 @@ async def pair(guild_id: int, timeblock: Timeblock):
         # Users may leave the server without unsubscribing
         # TODO: listen to that event and drop them from the table
         users = list(filter(None, users))
-        logger.info(
-            f"Pairing for G:{guild_id} T:{timeblock.name} with {len(users)}/{len(userids)} users."
-        )
+        await process_pairings(guild_id, users, timeblock, is_leetcode=False)
+
+        for difficulty in ["Easy", "Medium", "Hard"]:
+            leetcode_userids = leetcode_db.query_timeblock_difficulty(
+                guild_id, timeblock, difficulty
+            )
+            leetcode_users = [client.get_user(userid) for userid in leetcode_userids]
+            leetcode_users = list(filter(None, leetcode_users))
+            await process_pairings(
+                guild_id,
+                leetcode_users,
+                timeblock,
+                is_leetcode=True,
+                difficulty=difficulty,
+            )
+        # logger.info(f"Pairing for G:{guild_id} T:{timeblock.name} with {len(users)}/{len(userids)} users.")
```

Comment:
This was meant as an indicator for how many users `list(filter(None, users))` was filtering. We should add it back if we can.